awsecs/awsecs.go

```diff
@@ -166,10 +166,23 @@ func updateTaskDef(taskDefARN, gitsha, updateStrategy string, ecsClient ecsiface
 	previousGitsha := ""
 	shouldUpdate := false
 
-	// Update each container in task def to use same repo with new tag/sha
-	for i, container := range newTaskInput.ContainerDefinitions {
+	containersToUpdate := make(map[string]bool)
+	for _, c := range containers {
+		containersToUpdate[c] = true
+	}
+
+	// Update desired containers in task def to use same repo with new tag/sha
+	for i, containerDef := range newTaskInput.ContainerDefinitions {
+		// If service config does not specify which containers to update, we update all containers
+		// in that task def.
+		if len(containersToUpdate) != 0 {
+			if _, found := containersToUpdate[*containerDef.Name]; !found {
```

Comment:
Nit, can just do `found :=` since the zero value is returned if no entry exists which is `false`.